pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -32,6 +32,10 @@ type Limits interface {
 	DefaultLimits() *validation.Limits
 }
 
+type TenantBoundariesClient interface {
+	TenantInBoundaries(tenant string) bool
+}
+
```

Comment:
`TenantBoundaries` sounds confusing and does not give much context. What do you think about using function type instead of interface and calling it something like `IndexGatewayOwnsTenant func(tenant string) bool`?